Framework/DataHandling/src/LoadPLN.cpp

```diff
@@ -0,0 +1,963 @@
+
+#include "MantidDataHandling/LoadPLN.h"
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/Axis.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/LogManager.h"
+#include "MantidAPI/RegisterFileLoader.h"
+#include "MantidAPI/Run.h"
+#include "MantidDataHandling/LoadANSTOEventFile.h"
+#include "MantidDataObjects/EventWorkspace.h"
+#include "MantidGeometry/Instrument.h"
+#include "MantidGeometry/Instrument/ComponentInfo.h"
+#include "MantidGeometry/Instrument/DetectorInfo.h"
+#include "MantidGeometry/Instrument/RectangularDetector.h"
+#include "MantidGeometry/Objects/ShapeFactory.h"
+#include "MantidKernel/OptionalBool.h"
+#include "MantidKernel/PhysicalConstants.h"
+#include "MantidKernel/PropertyWithValue.h"
+#include "MantidKernel/UnitFactory.h"
+#include "MantidNexus/NexusClasses.h"
+
+#include <boost/filesystem.hpp>
+#include <boost/math/special_functions/round.hpp>
+#include <boost/math/tools/minima.hpp>
+
+#include <Poco/AutoPtr.h>
+#include <Poco/TemporaryFile.h>
+#include <Poco/Util/PropertyFileConfiguration.h>
+
+#include <algorithm>
+#include <cmath>
+#include <cstdio>
+#include <fstream>
+#include <iostream>
+
+namespace Mantid {
+namespace DataHandling {
+
+namespace {
+
+// number of physical detectors
+constexpr size_t MONITORS = 8;
+constexpr size_t DETECTOR_TUBES = 200;
+constexpr size_t HISTO_BINS_X = DETECTOR_TUBES + MONITORS;
+constexpr size_t HISTO_BINS_Y_DENUMERATOR = 16;
+constexpr size_t PIXELS_PER_TUBE = 1024 / HISTO_BINS_Y_DENUMERATOR;
+constexpr size_t DETECTOR_SPECTRA = DETECTOR_TUBES * PIXELS_PER_TUBE;
+constexpr size_t HISTOGRAMS = DETECTOR_SPECTRA + MONITORS;
+
+// File loading progress boundaries
+constexpr size_t Progress_LoadBinFile = 48;
+constexpr size_t Progress_ReserveMemory = 4;
+constexpr size_t Progress_Total =
+    2 * Progress_LoadBinFile + Progress_ReserveMemory;
+
+// Algorithm parameter names
+constexpr char FilenameStr[] = "Filename";
+constexpr char MaskStr[] = "Mask";
+constexpr char SelectDetectorTubesStr[] = "SelectDetectorTubes";
+constexpr char SelectDatasetStr[] = "SelectDataset";
+constexpr char FilterByTimeStartStr[] = "FilterByTimeStart";
+constexpr char FilterByTimeStopStr[] = "FilterByTimeStop";
+constexpr char PathToBinaryStr[] = "BinaryEventPath";
+constexpr char TOFBiasStr[] = "TimeOfFlightBias";
+constexpr char CalibrateTOFStr[] = "CalibrateTOFBias";
+
+// Common pairing of limits
+using TimeLimits = std::pair<double, double>;
+
+template <typename Type>
+void AddSinglePointTimeSeriesProperty(API::LogManager &logManager,
+                                      const std::string &time,
+                                      const std::string &name,
+                                      const Type value) {
+  // create time series property and add single value
+  auto p = new Kernel::TimeSeriesProperty<Type>(name);
+  p->addValue(time, value);
```

Comment:
Does your properties (times) have units?

Reply:
It uses the default (seconds) for time.

Reply:
When loading the unit test file, and investigating the sample logs, I do not see units (DetectorTankAngle, Wavelength, start_time, OverlapChopperFreq, dur, FermiChopperFreq, .. ?):
![samplelogs](https://user-images.githubusercontent.com/19187274/52778516-7cbb3b80-3046-11e9-96f3-82447ac49c06.png)

I struggled myself adding units everywhere in the sample logs (I have an issue open for this to attempt a general fix since getting a sample log property does not include getting the unit as well - this is manual work).

I understand, if this is not your priority but you may keep this in mind for future changes and optimisations.

Reply:
I will but will also be interested in the general fix when it is available. How would I know when a general fix is available?

Reply:
@geishm-ansto  I will keep you informed, did not yet work on this. The issue is https://github.com/mantidproject/mantid/issues/24633